Ciac 9741/venafi/v2 (#34139)
demisto/content: Packs/Venafi/Integrations/VenafiV2/README.md

    | --- | --- |
    | Server URL (e.g. https://192.168.0.1) | True |
    | User Name | True |
    | Password | True |
    | Client ID | True |
    | Trust any certificate (not secure) | False |
    | Use system proxy settings | False |

4. Click **Test** to validate the URLs, token, and connection.

## Commands

You can execute these commands from the Cortex XSOAR CLI, as part of an automation, or in a playbook.
After you successfully execute a command, a DBot message appears in the War Room with the command details.

### venafi-get-certificates

***
Gets Venafi certificates query. All dates are in 2016-11-12T00:00:00.0000000Z format. Additional fields information at https://ao-tlspd.dev.ven-eco.com/aperture/help/Content/SDK/WebSDK/r-SDK-Certificates-search-attribute.htm and https://ao-tlspd.dev.ven-eco.com/aperture/help/Content/SDK/WebSDK/r-SDK-Certificates-search-status.htm

#### Base Command

`venafi-get-certificates`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |
| CreatedOn | Specify YYYY-MM-DD or the ISO 8601 format. | Optional | 
| CreatedOnGreater | Specify YYYY-MM-DD or the ISO 8601 format. | Optional | 

> | CreatedOnLess | Specify YYYY-MM-DD or the ISO 8601 format. | Optional | 

Reviewer: But what is the definition of the argument?

CreatedOn - The date on which the certificated was created. Specify YYYY-MM-DD or the ISO 8601 format.
CreatedOnGreater - Find certificates created after this date. Specify YYYY-MM-DD or the ISO 8601 format.
CreatedOnLess - Find certificates created before this date. Specify YYYY-MM-DD or the ISO 8601 format.

| Disabled | Include only certificates that are enabled 0 or disabled 1. | Optional | 
| InError | Whether to include only certificates that are in an error state (1) or not in an error state (0). | Optional | 

> | ValidationState | Validation state. Possible values: "Blank", "Success", or "Failure". Possible values are: Blank, Success, Failure. | Optional | 

Reviewer: ```suggestion
| ValidationState | Validation state. Possible values are: Blank, Success, Failure. | Optional | 
```

> | ManagementType | Management type. Possible values: "Unassigned", "Monitoring", "Enrollment", or "Provisioning". Possible values are: Unassigned, Monitoring, Enrollment, Provisioning. | Optional | 

Reviewer: ```suggestion
| ManagementType | Management type. Possible values are: Unassigned, Monitoring, Enrollment, Provisioning. | Optional | 
```

| Name | Name of the certificate object. | Optional | 
| NetworkValidationDisabled | Whether to include only certificates with network validation disabled (1) or enabled (0). | Optional | 
| ParentDn | The full path to the parent of the object in Trust Protection Platform (e.g., \VED\Policy\Engineering,\VED\Policy\HR). | Optional | 
| ParentDnRecursive | The specific folder from which to retrieve certificates. (The subfolders will also be scanned.) Accepts a single value. | Optional | 

> | PendingWorkflow | Whether to include only certificates that are pending workflow resolution (have an outstanding workflow ticket). This parameter does not require a value to be specified. | Optional | 

Reviewer: ```suggestion
| PendingWorkflow | Whether to include only certificates that are pending workflow resolution (have an outstanding workflow ticket).  | Optional | 
```

| C | Find certificates by Country attribute of Subject DN. | Optional | 
| CN | Find certificates by Common name attribute of Subject DN. | Optional | 
| Issuer | Find certificates by issuer. Use the CN ,O, L, S, and C values from the certificate request. Surround the complete value within double quotes ("). If a value already has double quotes, escape them with a second set of double quotes. For example, OU=""(c) 2020 Entrust, Inc. - for authorized use only"". | Optional | 
| KeyAlgorithm | Algorithm for the public key in the certificate (e.g., RSA, DSA). | Optional | 
| KeySize | Comma-separated list of the bit size of the public key in the certificate (e.g., 2048). | Optional | 
| KeySizeGreater | The size for which the public key size is greater than. | Optional | 
| KeySizeLess | The size for which the public key size is less than. | Optional | 
| L | Find certificates by Locality/City attribute of Subject Distinguished Name (SDN). | Optional | 
| O | Find certificates by Organization attribute of Subject DN. | Optional | 
| S | Find certificates by State/Province attribute of Subject DN. | Optional | 

> | S | Find certificates by State/Province attribute of Subject DN. | Optional | 

Reviewer: Why is this twice?

| ValidTo | Date on which the certificate expires (e.g., 2015-10- 08T19:15:35.6431456Z or 2015-10-08). | Optional | 
| ValidToGreater | Date after which the certificates expire. | Optional | 
| ValidToLess | Date before which the certificates expire. | Optional | 
| Limit | The maximum number of certificates to return. Default value = 50. | Optional | 

#### Context Output

| **Path** | **Type** | **Description** |
| --- | --- | --- |
| Venafi.Certificate.CreatedOn | date | The exact date and time when the certificate object was created. | 
| Venafi.Certificate.DN | string | The DN of the certificate. | 
| Venafi.Certificate.Name | string | The name of the certificate. | 
| Venafi.Certificate.ParentDN | string | The full path to the parent of the object in Trust Protection Platform. | 
| Venafi.Certificate.SchemaClass | string | The class name of the certificate object. | 
| Venafi.Certificate.ID | string | The certificate object GUID. | 
| Venafi.Certificate.X509 | dictionary | Enrolled or issued certificate information: CN, Issuer, KeyAlgorithm, KeySize, SANS, Serial, Subject, Thumbprint, ValidFrom, ValidTo | 

### venafi-get-certificate-details

***

> Uses a certificate GUID to extract more details from the cert store.

Reviewer: ```suggestion
Uses a certificate GUID to extract more details from the certificate store.
```

#### Base Command

`venafi-get-certificate-details`

#### Input

| **Argument Name** | **Description** | **Required** |
| --- | --- | --- |

> | guid | The id of the certificate, add description that the user can get this value by running the command “venafi-get-certificates”. | Required | 

Reviewer: ```suggestion
| guid | The ID of the certificate. Add a description so the user can get this value by running the command “venafi-get-certificates”. | Required | 
```

| Description | string | Certificate description. | 
| ManagedBy | string | Certificate manager. | 
| ManagementType | string | The level of management that the Trust Protection Platform applies to the certificate. | 
| CertificateDetails.AIAKeyIdentifier | string | Authority key identifier. | 
| CertificateDetails.Issuer | string | The CN, O, L, S, and C values from the certificate request. | 
| CertificateDetails.Serial | string | The unique serial number that the CA assigned to the certificate. | 
| CertificateDetails.Subject | string | The CN, O, L, S, and C values from the certificate request. | 
| CertificateDetails.Thumbprint | string | The SHA1 thumbprint hash of the certificate. | 
| CertificateDetails.ValidFrom | string | Certificate validation start date. | 
| CertificateDetails.ValidTo | string | Certificate validation end time. | 
| CertificateDetails.AIACAIssuerURL | array | Available only when the certificate was issued by a well-configured CA. An array of Authority Information Access \(AIA\). Shows the CA issuer link and the CA's certificate details. May also include Online Certificate Status Protocol \(OCSP\) information about revocation. | 

> | CertificateDetails.CN | string | The Common name attribute of Subject Distinguished Name \(DN\). | 

Reviewer: ```suggestion
| CertificateDetails.CN | string | The Common Name attribute of Subject Distinguished Name \(DN\). | 
```

| CertificateDetails.KeySize | string | Only available for RSA certificates. The bit size of the public key. | 
| CertificateDetails.KeyUsage | string | A list of Key Usage extension values that describe the purpose of the public key. | 
| CertificateDetails.OU | string | An array of Organization Units or names. | 
| CertificateDetails.PublicKeyHash | string | The public key hash string. Available only when the certificate has a private key. | 
| CertificateDetails.SKIKeyIdentifier | string | The generated Subject Key Identifier \(SKI\) | 
| CertificateDetails.SignatureAlgorithm | string | The signature algorithm for signing the certificate. | 
| CertificateDetails.SignatureAlgorithmOID | string | The Signature Object ID for signing the certificate. | 
| CertificateDetails.StoreAdded | string | The Date Time stamp when the private key was added to the store. | 
| CertificateDetails.SubjectAltNameDNS | string | An array of Domain Name System \(DNS\) SANs. | 
| CertificateDetails.SubjectAltNameEmail | string | An array of Email SANs. Based on RFC 822. | 
| CertificateDetails.SubjectAltNameOtherNameUPN | string | An array of User Principal Name \(UPN\) SANs. | 

> | CertificateDetails.SubjectAltNameIPAddress | string | An array of IP Address SANs. | 

Reviewer: ```suggestion
| CertificateDetails.SubjectAltNameIPAddress | string | An array of IP address SANs. | 
```

| CertificateDetails.SubjectAltNameURI | string | An array of Uniform Resource Indicator \(URI\) SANs. | 
| CreatedBy | string | The object that initiated enrollment or provisioning changes. The default is Web SDK. | 

> | Origin | string |  | 

Reviewer: Need a description here.

| ProcessingDetails | dictionary | Absent when the certificate is not currently processing in the Trust Protection Platform lifecycle: InError, InProcess, Stage, Status, TicketDN | 
| RenewalDetails | dictionary | A list of certificate renewal information | 

> | ValidationDetails | dictionary | A list of host identity information and the overall certificate validation state result. If no validation occurred, only the lastvalidationstateupdate field appears. all other validationdetails fields are absent. | 

Reviewer: ```suggestion
| ValidationDetails | dictionary | A list of host identity information and the overall certificate validation state result. If no validation occurred, only the lastvalidationstateupdate field appears. All other validationdetails fields are absent. | 
```
